src/sentry/integrations/slack/notify_action.py

```diff
@@ -182,16 +182,22 @@ def get_channel_id(self, integration_id, name):
             'exclude_members': True,
         })
 
-        resp = session.get('https://slack.com/api/channels.list', params=channels_payload)
-        resp = resp.json()
-        if not resp.get('ok'):
-            self.logger.info('rule.slack.channel_list_failed', extra={'error': resp.get('error')})
-            return None
+        # Slack limits the response of `channels.list` to 1000 channels, paginate if needed
+        cursor = None
+        while cursor or len(channels) == 0:
```

Comment:
Where is `channels` coming from here?

Reply:
Sorry, that was from a bad merge. I resolved this concern and put up a new PR https://github.com/getsentry/sentry/pull/11064